### ISSUE TITLE
[Feature Request] Improve auth guard

### ISSUE THREAD

**Issue by bartvanb** [vantage6/vantage6-UI#142]
**Problem description**
The authorization guard currently only checks if a user is authenticated, not if they are allowed to acccess a page. This leads to errors if users manually type a URL that they are not allowed access to.

**Desired solution**
Check if user is authorized to a certain page. If not, send them back to previous page.

**Comment by eutj** [comment]
How can the UI determine the authorization for the current user for a specific page?

**Comment by eutj** [comment]
You mean scope=GLOBAL?

**Comment by bartvanb** [comment]
Nope, with any scope they would be able to view the page. The scope just determines how much they get to see: if they have `global` scope they will get to see everything, if `organization` just their own organization, if `collaboration` all organizations in their collaborations.

**Comment by richard-elastique** [comment]
The auth guards should basically use the same checks that are already present for the UI elements that users use to navigate to these pages.

Would be a nice code quality feature if there is only one single truth for the values needed to make these checks.

**Comment by bartvanb** [comment]
After a discussion with Eugene, I gave this issue low priority: for now, we can just go to a page and the API will return an error message if the user is not allowed to access the resources, which should be displayed in general error handling.
